graylog2-server/src/main/java/org/graylog2/indexer/counts/Counts.java

```diff
@@ -45,6 +45,12 @@ public long total() {
     public long total(final IndexSet indexSet) {
         final String[] names = indexSet.getManagedIndicesNames();
 
+        // Return 0 if there are no indices in the given index set. If we run the query with an empty index list,
+        // Elasticsearch will count all documents in all indices and thus return a wrong count.
+        if (names.length == 0) {
```

Comment:
This check should probably also be added to the `Counts#total()` method.

If none of the index set contains an index and other indices (not managed by Graylog) exist in the Elasticsearch cluster, the returned value might be wrong.